src/app/games/connect-six/tests/ConnectSixRules.spec.ts

```diff
@@ -21,14 +23,29 @@ describe('ConnectSixRules', () => {
     const X: PlayerOrNone = PlayerOrNone.ONE;
 
     let rules: ConnectSixRules;
+    const defaultConfig: MGPOptional<GobanConfig> = ConnectSixRules.get().getDefaultRulesConfig();
 
     beforeEach(() => {
         rules = ConnectSixRules.get();
     });
+
     describe('first turn', () => {
+
+        it('should not create move when coord is out of board', () => {
+            // Given the initial state
+            const state: ConnectSixState = ConnectSixRules.get().getInitialState(defaultGobanConfig);
+
+            // When dropping out of the board
+            const move: ConnectSixMove = ConnectSixFirstMove.of(new Coord(-1, -1)) as ConnectSixMove;
+
+            // Then it should be illegal
+            const reason: string = CoordFailure.OUT_OF_RANGE(new Coord(-1, -1));
+            RulesUtils.expectMoveFailure(rules, state, move, reason, defaultConfig);
+        });
+
         it('should allow the first player play only one piece', () => {
             // Given the initial state
-            const state: ConnectSixState = ConnectSixRules.get().getInitialState();
+            const state: ConnectSixState = ConnectSixRules.get().getInitialState(defaultGobanConfig);
```

Comment:
idem, don't use defaultGobanConfig if you have defaultConfig

Reply:
voir plus haut !

Reply:
voir plus haut !